### PR TITLE
executor: refactor info from interface to string to avoid runTime.convTstring cost

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve? 

tiny refactor to reduce interface alloc/copy cost

```
  790ms      4.43s (flat, cum)  1.88% of Total
         .          .    959:	s.parser.SetSQLMode(s.sessionVars.SQLMode)
         .          .    960:	s.parser.EnableWindowFunc(s.sessionVars.EnableWindowFunction)
         .          .    961:	return s.parser.Parse(sql, charset, collation)
         .          .    962:}
         .          .    963:
     140ms      140ms    964:func (s *session) SetProcessInfo(sql string, t time.Time, command byte, maxExecutionTime uint64) {
         .          .    965:	var db interface{}
      20ms       20ms    966:	if len(s.sessionVars.CurrentDB) > 0 {
      20ms      1.13s    967:		db = s.sessionVars.CurrentDB
         .          .    968:	}
         .          .    969:
         .          .    970:	var info interface{}
         .          .    971:	if len(sql) > 0 {
      30ms      130ms    972:		info = sql
         .          .    973:	}
      20ms      1.30s    974:	pi := util.ProcessInfo{
         .          .    975:		ID:               s.sessionVars.ConnectionID,
      10ms       10ms    976:		DB:               db,
         .          .    977:		Command:          command,
      80ms       80ms    978:		Plan:             s.currentPlan,
         .          .    979:		Time:             t,
         .          .    980:		State:            s.Status(),
      10ms       10ms    981:		Info:             info,
     240ms      240ms    982:		CurTxnStartTS:    s.sessionVars.TxnCtx.StartTS,
         .          .    983:		StmtCtx:          s.sessionVars.StmtCtx,
      20ms       20ms    984:		StatsInfo:        plannercore.GetStatsInfo,
      60ms      290ms    985:		MaxExecutionTime: maxExecutionTime,
         .          .    986:	}
         .          .    987:	if s.sessionVars.User != nil {
     120ms      130ms    988:		pi.User = s.sessionVars.User.Username
         .          .    989:		pi.Host = s.sessionVars.User.Hostname
         .          .    990:	}
      10ms      920ms    991:	s.processInfo.Store(&pi)
      10ms       10ms    992:}
```

into

```
 10ms       10ms    966:	if len(s.sessionVars.CurrentDB) > 0 {
         .          .    967:		db = s.sessionVars.CurrentDB
         .          .    968:	}
         .          .    969:
         .          .    970:	var info string
         .          .    971:	if len(sql) > 0 {
         .          .    972:		info = sql
         .          .    973:	}
      30ms      1.40s    974:	pi := util.ProcessInfo{
      80ms       80ms    975:		ID:               s.sessionVars.ConnectionID,
         .          .    976:		DB:               db,
         .          .    977:		Command:          command,
     120ms      120ms    978:		Plan:             s.currentPlan,
         .          .    979:		Time:             t,
      20ms       20ms    980:		State:            s.Status(),
         .          .    981:		Info:             info,
     170ms      170ms    982:		CurTxnStartTS:    s.sessionVars.TxnCtx.StartTS,
      10ms       10ms    983:		StmtCtx:          s.sessionVars.StmtCtx,
         .          .    984:		StatsInfo:        plannercore.GetStatsInfo,
         .          .    985:		MaxExecutionTime: maxExecutionTime,
         .          .    986:	}
```

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

Related changes

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -962,12 +962,12 @@ func (s *session) ParseSQL(ctx context.Context, sql, charset, collation string)
 }
 
 func (s *session) SetProcessInfo(sql string, t time.Time, command byte, maxExecutionTime uint64) {
-	var db interface{}
+	var db string
```
Comment (ngaut) [inline session/session.go]: Remove the follow 9 lines, you can simply use s.sessionVars.CurrentDB and sql inside the pi assignment.

File: util/processinfo.go [hunk 2]
```diff
@@ -45,19 +45,23 @@ type ProcessInfo struct {
...
 		if full {
-			info = pi.Info.(string)
+			info = pi.Info
 		} else {
-			info = fmt.Sprintf("%.100v", pi.Info.(string))
+			info = fmt.Sprintf("%.100v", pi.Info)
 		}
 	}
 	t := uint64(time.Since(pi.Time) / time.Second)
+	var db interface{}
```
Comment (ngaut) [inline util/processinfo.go]: Remove the follow 4 lines, use pi.DB instead of db

**Comment by zz-jason** [comment]
@wshwsh12 PTAL. I remembered that these two fields are set to `interface` intentionally, in order to display the `NULL` value.

**Comment by cfzjywxk** [comment]
> @wshwsh12 PTAL. I remembered that these two fields are set to `interface` intentionally, in order to display the `NULL` value.

ToRowForShow here will transfer process into into row formats(interface array),  nil element will result into NULL value

File: util/processinfo.go (hunk 2 above)
Comment (cfzjywxk) [reply util/processinfo.go]: ToRowForShow here will transfer process into into row formats(interface array), nil interface is needed to represent "NULL" column value

**Comment by zz-jason** [comment]
> > @wshwsh12 PTAL. I remembered that these two fields are set to `interface` intentionally, in order to display the `NULL` value.
> 
> ToRowForShow here will transfer process into into row formats(interface array), nil element will result into NULL value

Got it.

**Pull request merged**